sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala

```diff
@@ -88,12 +88,24 @@ class CacheManager extends Logging with AdaptiveSparkPlanHelper {
       query: Dataset[_],
       tableName: Option[String] = None,
       storageLevel: StorageLevel = MEMORY_AND_DISK): Unit = {
-    val planToCache = query.logicalPlan
+    cacheQueryWithLogicalPlan(query.sparkSession, query.logicalPlan, tableName, storageLevel)
+  }
+
+  /**
+   * Caches the data produced by the given [[LogicalPlan]].
+   * Unlike `RDD.cache()`, the default storage level is set to be `MEMORY_AND_DISK` because
+   * recomputing the in-memory columnar representation of the underlying table is expensive.
+   */
+  def cacheQueryWithLogicalPlan(
+      spark: SparkSession,
+      planToCache: LogicalPlan,
+      tableName: Option[String] = None,
```

Comment:
does this method need default parameter values?

Reply:
I think https://github.com/apache/spark/pull/30815/files#r544822268 should work. Let me update this PR. Thanks!